kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -159,7 +173,9 @@ public KeyValueIterator<K, Result<K>> backAll(
   }
 
   long offset() {
-    return client.getOffset(tableName, partition).offset;
+    // since all subpartitions use the same offset, we can just use the
+    // first subpartition to get the offset
+    return client.getOffset(tableName, subPartitioner.first(partition)).offset;
```

Comment:
the reasoning here is that this is safe because the following two statements are true:
1. an offset is only written to _any_ subpartition iff _all_ subpartitions have successfully been flushed up to that offset
2. it is safe to restore from any offset up to and including the flushed offset

correct?

Reply:
I actually think we need to do both - otherwise the permit revocation might break (we only revoke permits when our offset is >= the offset in the ongoing transaction), so if partition `0` is less than the committed offset for `1` and I used the offset for `0` to do the revocation then `1` wouldn't get revoked and we run into the same issue.